### PR TITLE
AK+Everywhere: Add new failable JsonArray::{append/set} functions

### PR DESCRIPTION
**Pull request by cammo1123**
Quite the Yakhole was explored here. Initially started off trying to `Convert JsonArrayModel to use Core::File` but these changes made it a lot easier to do so.

### CODE REVIEW

**Review by GMTA — CHANGES_REQUESTED** [review]

File: AK/JsonArray.h [hunk 1]
```diff
@@ -61,9 +61,12 @@ class JsonArray {
 
     [[nodiscard]] JsonValue take(size_t index) { return m_values.take(index); }
 
+    void must_append(JsonValue value) { m_values.append(move(value)); }
+    void must_set(size_t index, JsonValue value) { m_values.insert(index, move(value)); }
```
[the remark this answers is not in the document]
Comment (cammo1123) [reply AK/JsonArray.h:65]: Doesn't it already do that internally? Or should i `MUST` it manually?

https://github.com/SerenityOS/serenity/blob/f0dde6236d8fbd1ae17e33f868df139467872768/AK/Vector.h#L257-L269

File: AK/JsonArray.h [hunk 2]
```diff
@@ -61,9 +61,12 @@ class JsonArray {
...
+
     void clear() { m_values.clear(); }
-    void append(JsonValue value) { m_values.append(move(value)); }
-    void set(size_t index, JsonValue value) { m_values[index] = move(value); }
+    ErrorOr<void> append(JsonValue value) { return TRY(m_values.try_append(move(value))); }
+    ErrorOr<void> set(size_t index, JsonValue value) { return TRY(m_values.try_insert(index, move(value))); }
```
[the remark this answers is not in the document]
Comment (cammo1123) [reply AK/JsonArray.h]: Oh! didn't know that :^)

File: AK/JsonArray.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (GMTA) [reply AK/JsonArray.h:65]: I could have sworn I read `try_*` methods there, you are right! These don't need additional `MUST`s :-)

File: Userland/Applications/BrowserSettings/BrowserSettingsWidget.h [hunk 3]
```diff
@@ -15,13 +15,14 @@
 class BrowserSettingsWidget final : public GUI::SettingsWindow::Tab {
     C_OBJECT(BrowserSettingsWidget)
```
[the remark this answers is not in the document]
Comment (cammo1123) [reply Userland/Applications/BrowserSettings/BrowserSettingsWidget.h]: Learning lots of things!

File: Userland/Applications/BrowserSettings/BrowserSettingsWidget.cpp [hunk 4]
```diff
@@ -57,10 +58,18 @@ class ColorSchemeModel final : public GUI::Model {
     Vector<ColorScheme> m_color_schemes;
 };
 
-BrowserSettingsWidget::BrowserSettingsWidget()
+ErrorOr<NonnullRefPtr<BrowserSettingsWidget>> BrowserSettingsWidget::create()
 {
-    load_from_gml(browser_settings_widget_gml).release_value_but_fixme_should_propagate_errors();
+    auto widget = adopt_ref(*new BrowserSettingsWidget);
```
[the remark this answers is not in the document]
Comment (cammo1123) [reply Userland/Applications/BrowserSettings/BrowserSettingsWidget.cpp]: Learning lots of things :^)

**Comment by AtkinsSJ** [comment]
This has conflicts.

**Pull request merged**